### PR TITLE
add option to disable pipeline partitioning

### PR DESCRIPTION
**Pull request by nelyahu**
added pipeline configuration to deepspeed config under pipeline section pipe_partitioned = "auto"
grad_partitioned = "auto"
can be used to enable or disable activations and grads partitioning in TP/PP mode

### CODE REVIEW

**Comment by nelyahu** [comment]
@ShadenSmith @jeffra @duli2012 @tjruwase - can you please review?

**Comment by nelyahu** [comment]
@tjruwase Hi :) can someone help reviewing this change?

File: deepspeed/runtime/config.py [hunk 1]
```diff
@@ -446,6 +446,8 @@ def get_pipeline_config(param_dict):
         "partition": "best",
         "seed_layers": False,
         "activation_checkpoint_interval": 0,
+        "pipe_partitioned": "auto",
```
Comment (tjruwase) [inline deepspeed/runtime/config.py:449]: Why use "auto" as a value to represent True rather than simply make this a boolean config?

Comment (nelyahu) [reply deepspeed/runtime/config.py]: @tjruwase agreed, see updated change

**Review by tjruwase — APPROVED** [review]

**Review by loadams — APPROVED** [review]

**Comment by loadams** [comment]
@tjruwase - should we add this to any docs we have?

**Comment by tjruwase** [comment]
> @tjruwase - should we add this to any docs we have?

@loadams, yes, we should. But I don't believe we have any docs for DeepSpeed pipeline parallelism support.

**Pull request merged**